src/main/java/com/artipie/front/api/ApiAuthFilter.java

```diff
@@ -0,0 +1,86 @@
+/*
+ * The MIT License (MIT) Copyright (c) 2022 artipie.com
+ * https://github.com/artipie/front/LICENSE.txt
+ */
+package com.artipie.front.api;
+
+import java.time.Instant;
+import java.util.Optional;
+import javax.json.Json;
+import org.eclipse.jetty.http.HttpStatus;
+import spark.Filter;
+import spark.Request;
+import spark.Response;
+import spark.Spark;
+
+/**
+ * Spark filter for API authentication.
+ * @since 1.0
+ */
+public final class ApiAuthFilter implements Filter {
+
+    /**
+     * Auth token validator.
+     */
+    private final TokenValidator validator;
+
+    /**
+     * New API auth filter.
+     * @param validator Token validator
+     */
+    public ApiAuthFilter(final TokenValidator validator) {
+        this.validator = validator;
+    }
+
+    @Override
+    public void handle(final Request request, final Response response) throws Exception {
+        try {
+            final var uid = this.validator.validate(
+                Optional.of(request.headers("Authorization")).orElse(""),
+                Instant.now()
+            );
+            request.attribute("uid", uid);
+        } catch (final ValidationException vex) {
+            Spark.halt(
+                HttpStatus.UNAUTHORIZED_401,
+                Json.createObjectBuilder()
+                    .add("error", vex.getMessage())
```

Comment:
if I'm not mistaken, `getMessage()` is possible nullable method, so I'd suggest to handle this case here and do not return `error: null` in any cases

Reply:
We require the message to be provided in the exception constructor parameter, so the only way to get `null` here is if it's passed by our code as `new ValidationException(null)`, but this is not allowed by static analyzer.